src/lib/command_line.ml

```diff
@@ -80,10 +81,25 @@ module Server = struct
     in
     Log.log t.log ~section:["server"; subsection] json
 
-  let save_job_list t =
-    Storage.Json.save_jsonable t.storage
-      ~path:["server"; "jobs.json"]
-      (`List (List.map t.jobs ~f:(fun j -> `String (Job.id j))))
+  let change_job_list t action =
+    let mutex =
+      match t.job_list_mutex with
+      | Some s -> s
+      | None ->
+        let m = Lwt_mutex.create () in
```

Comment:
Shouldn't this be initialized at server init time? This looks like a potential race condition (two calls to this fn could result in the first call to hold an invalid, overwritten mutex)? I don't know how this is called though, so maybe it's never called in parallel (but then why a mutex?) Why do we need a mutex?